engine/websocketroutine_manager.go

```diff
@@ -135,7 +136,7 @@ func (m *WebsocketRoutineManager) websocketRoutine() {
 				}
 
 				if ws.IsEnabled() {
-					err = ws.Connect()
+					err = ws.Connect(context.TODO(), false /*this will not auto-subscribe to websocket subscriptions*/)
```

Comment:
This is a change in behaviour. If the websocket disconnects and reconnects, we will no longer be able to resubscribe automatically via the connection manager. Eg Binance:

```
[INFO]  | SYNC | 05/09/2023 11:34:47 | Binance websocket BTC-USDT SPOT TICKER: Last 25792.86000000 Ask 25792.87000000 Bid 25792.86000000 High 26108.02000000 Low 25631.21000000 Volume 22165.03711000
[WARN]  | WEBSOCKET | 05/09/2023 11:34:47 | Binance websocket has been disconnected. Reason: read tcp 192.168.1.21:49606->18.176.86.233:9443: read: socket is not connected
[DEBUG] | WEBSOCKET | 05/09/2023 11:34:47 | Binance websocket: running connection monitor cycle
[ERROR] | WEBSOCKET | 05/09/2023 11:34:49 | Binance Error connecting Binance - Unable to connect to Websocket. Error: Binance websocket connection: wss://stream.binance.com:9443/stream Error: dial tcp 54.249.21.20:9443: connect: bad file descriptor
[DEBUG] | WEBSOCKET | 05/09/2023 11:34:51 | Binance websocket: running connection monitor cycle
[INFO]  | WEBSOCKET | 05/09/2023 11:34:54 | Binance Websocket connected to wss://stream.binance.com:9443/stream
[DEBUG] | WEBSOCKET | 05/09/2023 11:34:54 | Binance websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 05/09/2023 11:34:56 | Binance websocket: running connection monitor cycle
[DEBUG] | ORDER | 05/09/2023 11:34:58 | Finished processing orders
[DEBUG] | WEBSOCKET | 05/09/2023 11:34:58 | Binance websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 05/09/2023 11:35:00 | Binance websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 05/09/2023 11:35:02 | Binance websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 05/09/2023 11:35:04 | Binance websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 05/09/2023 11:35:06 | Binance websocket: running connection monitor cycle
```

Now on `master`
```
[DEBUG] | WEBSOCKET | 05/09/2023 11:36:47 | Binance websocket connection: message received: {"stream":"btcusdt@trade","data":{"e":"trade","E":1693877807105,"s":"BTCUSDT","t":3205834858,"p":"25785.02000000","q":"0.00178000","b":22237248997,"a":22237248025,"T":1693877807105,"m":false,"M":true}}
[WARN]  | WEBSOCKET | 05/09/2023 11:36:47 | Binance websocket has been disconnected. Reason: read tcp 192.168.1.21:49640->52.68.13.86:9443: read: socket is not connected
[DEBUG] | WEBSOCKET | 05/09/2023 11:36:48 | Binance websocket: running connection monitor cycle
[ERROR] | WEBSOCKET | 05/09/2023 11:36:50 | Binance Error connecting Binance - Unable to connect to Websocket. Error: Binance websocket connection: wss://stream.binance.com:9443/stream Error: dial tcp 52.193.119.30:9443: connect: bad file descriptor
[DEBUG] | WEBSOCKET | 05/09/2023 11:36:50 | Binance websocket: running connection monitor cycle
[INFO]  | WEBSOCKET | 05/09/2023 11:36:53 | Binance Websocket connected to wss://stream.binance.com:9443/stream
[DEBUG] | WEBSOCKET | 05/09/2023 11:36:53 | Binance websocket connection: sending message to websocket {Method:SUBSCRIBE Params:[btcusdt@ticker btcusdt@trade btcusdt@kline_1m btcusdt@depth@100ms] ID:0}
```

Reply:
I'm fine with the ability to decide these things via external management, but I'm not seeing a good reason for this change in default behaviour and I don't think you made this change intentionally. You may need extra nuance in this feature.

Reply:
Reverted